### ISSUE TITLE
Migrate to the public API

### ISSUE THREAD

**Issue by klanchman** [klanchman/homebridge-logi-circle-controls#17]
A Logitech employee reached out about the plugin. It turns out Logitech now has a [public API](https://developers.logitech.com/circle) for the Logi Circle! The public documentation went up about 6 months ago.

Migrating to the public API will make me more confident in any new feature development since I won't be relying on an undocumented API meant only for Logitech's internal use.

There are going to be a few things to do here. I'll check things off as I get them done.

- [ ] Use the OAuth flow for authentication
- [ ] Change all API calls to work with the public API
- [ ] Create documentation for users on how to set up OAuth

I'll probably make at least 2 approaches for the OAuth documentation: one where the user exposes the plugin to the Internet, and another where the user manually intervenes in the auth flow.

---

References:

- [API Documentation](https://developers.logitech.com/circle)
- [PR for Home Assistant to switch to the public API](https://github.com/home-assistant/home-assistant/pull/20624)
- [Wink Homebridge plugin that uses OAuth](https://github.com/sibartlett/homebridge-wink3/blob/master/src/WinkClient.js)
- [Home Assistant instructions for configuring their integration](https://www.home-assistant.io/integrations/logi_circle)
- [Home Assistant forum thread for the integration, including discussion about OAuth redirect URIs](https://community.home-assistant.io/t/logi-circle-camera-with-hass/19761/57)
- [Home Assistant GitHub issue regarding redirect URIs](https://github.com/evanjd/hass-logi-circle/issues/3#issuecomment-488922238)

**Comment by Nastras** [comment]
Hey, @klanchman

did Aron from Logitech write to you?

I changed the days of my email in my Logitech account and forgot to change it in my config.
This caused the plugin to send a lot of incorrect requests to the API. I was contacted by Aron
notifies that I have made a mistake. I noticed that I forgot to change the mail address in the config.

I told him about your plugin and homebridge and he told me that you don't use the public api and that there is a public one.

I wanted to tell you that but unfortunately I didn't get to it.

I see you already have the info and I am happy that it is helpful for you to develop the plugin further.


Best regards

Nastra

**Comment by klanchman** [comment]
Hi @Nastras, yes Aron was the one who contacted me 🙂

In addition to this ticket I also did #16 to help prevent the plugin from sending too many bad requests. That’s available in version 2.1.2 of the plugin now!

**Comment by Nastras** [comment]
I hope you're not mad. Now that you've got to work because of my mistake?

😅😅😅

**Comment by klanchman** [comment]
Not at all! I actually thought I had accounted for a situation like this, but I realized that I messed up, and login was not covered by that. And I learned about the public API from it, which I’m really happy about 😄

**Comment by Nastras** [comment]
Then I'm glad. 😀👍👍👍

**Comment by klanchman** [comment]
I reached out to Logitech in December about some of this, but haven't heard anything. I filled out their form to get an API key in January, but haven't heard anything.

Based on what I've seen about the Logi Circle plugin for Home Assistant[^1], I don't exactly have high hopes that they'll get back to me any time soon. That's worrisome, since anyone who used this plugin would need to go through the process of getting an API key as well. Logitech told me about plans for a self-service portal for getting keys, but it doesn't exist yet (unfortunately I don't think it's likely ever to come to fruition).

For the foreseeable future I'm going to keep using the internal API. Even if Logitech were to set me up with keys soon, I don't want to send users through the same process 😬 

***

[^1]: The maintainer of the Home Assistant plugin talks about some of his experience [in this comment](https://github.com/home-assistant/core/issues/36521#issuecomment-747892879), and you can find more discussion peppered around in [their forum thread](https://community.home-assistant.io/t/logi-circle-camera-with-hass/19761).
